File: Lpp.Adapters/Lpp.Dns.DataMart.Model.Interface/Rest/StatusCode.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Runtime.Serialization;
using System.Text;

namespace Lpp.Dns.DataMart.Model.Rest
{
    /// <summary>
    /// Corresponds to IModelProcessor's RequestStatus.StatusCode enum.
    /// </summary>
    [DataContract(Namespace = "http://lincolnpeak.com/schemas/DNS4/API")]
    public enum StatusCode
    {
        [EnumMember]
        Complete = 66,            // 1000010
        [EnumMember]
        CompleteWithMessage = 67, // 1000011
        [EnumMember]
        Pending = 4,              // 0000100
        [EnumMember]
        InProgress = 8,           // 0001000
        [EnumMember]
        AwaitingApproval = 16,    // 0010000
        [EnumMember]
        Canceled = 80,            // 1010000
        [EnumMember]
        Error = 96,               // 1100000
    }
}

File: PopMedNet.vNext/Plugins/Lpp.Dns.HealthCare.I2B2RDP/PMNServerAdapter/src/com/lincolnpeak/i2b2/restlet/pmnresources/StatusCode.java
package com.lincolnpeak.i2b2.restlet.pmnresources;

import javax.xml.bind.annotation.XmlEnum;

@XmlEnum(String.class)
public enum StatusCode
{
    Complete(66),            // 1000010
    CompleteWithMessage(67), // 1000011
    Pending(4),              // 0000100
    InProgress(8),           // 0001000
    AwaitingApproval(16),    // 0010000
    Canceled(80),            // 1010000
    Error(96);               // 1100000

    private int code;
    
    StatusCode(int code)
    {
    	this.code = code;
    }
    
    StatusCode()
    {
    }
    
    public void setCode(int code)
    {
    	this.code = code;
    }
    
    public int getCode()
    {
    	return code;
    }
}